cg/meta/transfer/flowcell.py

```diff
@@ -20,111 +46,154 @@ def __init__(self, db: Store, stats_api: StatsAPI, hk_api: HousekeeperAPI):
+        sample_sheet_path: Path = Path(
+            flow_cell_dir, DemultiplexingDirsAndFiles.SAMPLE_SHEET_FILE_NAME
+        )
+        self._include_sequencing_file(
+            file_path=sample_sheet_path,
+            flow_cell_id=flow_cell_id,
+            store=store,
+            tag_name=SequencingFileTag.SAMPLE_SHEET,
+        )
+
+    def _include_cgstats_log_to_housekeeper(
+        self, flow_cell_dir: Path, flow_cell_id: str, store: bool
+    ) -> None:
+        """Include cgstats log in Housekeeper."""
+        for cgstats_log_path in flow_cell_dir.glob("stats-*[0-9]-*"):
+            self._include_sequencing_file(
+                file_path=cgstats_log_path,
+                flow_cell_id=flow_cell_id,
+                store=store,
+                tag_name=SequencingFileTag.CGSTATS_LOG,
+            )
+
+    def _store_sequencing_files(
+        self,
+        flow_cell_id: str,
+        sequencing_files: List[str],
+        tag_name: str,
+        sample_id: Optional[str] = None,
+    ) -> None:
+        """Stor sequencing file(s) in Housekeeper."""
+        bundle_name: str = sample_id if sample_id else flow_cell_id
+        hk_bundle: Bundle = self.hk.bundle(bundle_name)
         if hk_bundle is None:
-            hk_bundle = self.hk.new_bundle(flow_cell_id)
-            self.hk.add_commit(hk_bundle)
-            new_version = self.hk.new_version(created_at=hk_bundle.created_at)
-            hk_bundle.versions.append(new_version)
-            self.hk.commit()
-            LOG.info(f"added new Housekeeper bundle: {hk_bundle.name}")
+            self.hk.create_new_bundle_and_version(name=bundle_name)
 
         with self.hk.session_no_autoflush():
-            hk_version = hk_bundle.versions[0]
-            if self.hk.files(path=sample_sheet_path).first() is None:
-                LOG.info(f"Adding samplesheet: {sample_sheet_path}")
-                tags = [self.hk.tag("samplesheet"), self.hk.tag(flow_cell_id)]
-                new_file = self.hk.new_file(path=sample_sheet_path, tags=tags)
-                hk_version.files.append(new_file)
-        self.hk.commit()
-
-    def _sample_sheet_path(self, flow_cell_id: str) -> str:
-        """Construct the path to the samplesheet to be stored."""
-        run_name: str = self.stats.run_name(flow_cell_id)
-        document_path: str = self.stats.document_path(flow_cell_id)
-        unaligned_dir: str = Path(document_path).name
-        root_dir: Path = self.stats.root_dir
-        return str(root_dir.joinpath(run_name, unaligned_dir, "SampleSheet.csv"))
+            for file in sequencing_files:
+                if self.hk.files(path=file).first() is None:
+                    LOG.info(f"Found new file: {file}.")
+                    LOG.info(f"Adding file using tag: {tag_name}")
+                    self.hk.add_and_include_file_to_latest_version(
+                        bundle_name=bundle_name, file=Path(file), tags=[tag_name, flow_cell_id]
```

Comment:
It would complicate the flow of the transfer function since I would have to move stuff around. I do not disagree with the suggestion, but it deserves a PR of its own.